engines/dm/menus.cpp

```diff
@@ -1628,8 +1628,11 @@ void MenuMan::printMessageAfterReplacements(const char *str) {
 				replacementString = championMan._champions[_vm->ordinalToIndex(championMan._actingChampionOrdinal)]._name;
 
 			*curCharacter = '\0';
-			strcat(outputString, replacementString);
-			curCharacter += strlen(replacementString);
+			size_t ln = Common::strlcat(outputString, replacementString);
+			if (ln >= sizeof(outputString)) {
```

Comment:
Interesting solution, what is going on here? What is the side effect of ending execution here?

Reply:
Indeed, I made it an error. It's cleaner.